src/py42/clients/legalhold.py

```diff
@@ -104,7 +108,10 @@ def get_matter_by_uid(self, legal_hold_uid):
             :class:`py42.response.Py42Response`: A response containing the Matter.
         """
         uri = u"/api/LegalHold/{}".format(legal_hold_uid)
-        return self._session.get(uri)
+        try:
+            return self._session.get(uri)
+        except (Py42BadRequestError, Py42ForbiddenError) as err:
+            raise Py42LegalHoldNotFoundOrPermissionDeniedError(err, legal_hold_uid)
```

Comment:
Wouldn't it be nice if we separate this into two exceptions. Something like:
```
except Py42BadRequestError ....:
   raise Py42LegalHoldNofFound ...
except Py42ForbiddenError ...:
  raise Py42PermissionDeniedError ...
```

 having a `Py42PermissionDeniedError` error might be useful in future. Since our sdk starts with authentication this case doesn't apply much but if there is some API having a role based feature, having "permission denied error" can be useful. 
Quoting "Zen of Python": Explicit is better than implicit.

Reply:
I think `Py42ForbiddenError` is `Py42PermissionDeniedError`, but I agree these should be split. I assume this s a common use case for users not to have access to legal hold, and therefore we made this exception to increase usability there. @timabrmsn Am I miss anything there? Do you foresee any issue? In the CLI, we will just catch both legal hold-specific exceptions.  Is this 1-1 meaning `Py42BadRequest` actually maps to `Py42LegalholdNotFound`?

Reply:
I did a bit of testing on this, and it looks like the Py42BadRequest exception is raised when the hold_uid is either not an integer, or else too large (look like 20 digits long is the cutoff). Otherwise it raises the Py42ForbiddenError if the ID doesn't exist or we don't have permission to view it.

Reply:
Ok... Well, that explains why it was the way it was. I will put it back - but I will add a comment explaining this so that if we cross this bridge again, we will remember these findings. thanks

Reply:
Or would it be better to leave bad request as it is and this catch `Py42PermissionDeniedError` and have a similar message to be, like "The ID may not exist or you may not have permission to view the matter" ?

Reply:
https://github.com/code42/py42/pull/159/commits/5ad1f3f74300709ab912645f6d944a5f836c1295

Reply:
Given that we have documented as explained by @timabrmsn above, as long as the user can look into the exception and figure out  which error occurred its fine if we keep it as is.

Reply:
I made it such that only Forbidden raises the error